ui/js/directives/clonejobs.js

```diff
@@ -81,21 +81,22 @@ treeherder.directive('thCloneJobs', [
             };
         }
 
-        jobs = $(selector);
-        var idx = jobs.index(jobs.filter(".selected-job"));
-        idx = getIndex(idx, jobs);
+        jobs = $(selector).filter(":visible");
+        if (jobs.length) {
+            var selIdx = jobs.index(jobs.filter(".selected-job"));
+            var idx = getIndex(selIdx, jobs);
 
-        el = $(jobs[idx]);
-        while (el.css('display') === 'none') {
-            idx = getIndex(idx, jobs);
             el = $(jobs[idx]);
+            key = el.attr(jobKeyAttr);
+            if (jobMap && jobMap[key] && selIdx !== idx) {
+                selectJob(jobMap[key].job_obj);
+                return;
+            }
         }
-
-        key = el.attr(jobKeyAttr);
-        if (jobMap && jobMap[key]) {
-            selectJob(jobMap[key].job_obj);
-        }
-
+        // if there was no new job selected, then ensure that we clear any job that
+        // was previously selected.
+        $timeout($rootScope.closeJob, 0);
```

Comment:
This is a really good point.  If think if the job is not visible (it's been classified and therefore filtered away) then unselecting it is the right call.  But if the job is still visible, then it should remain selected.

Great catch!!  Thanks!